src/catchup/CatchupWork.cpp

```diff
@@ -392,9 +396,9 @@ CatchupWork::runCatchupStep()
                     return mApplyBufferedLedgersWork->getState();
                 }
             }
-            // see if we need to apply buffered ledgers
-            if (mApp.getCatchupManager().hasBufferedLedger())
+            else if (mApp.getCatchupManager().hasBufferedLedger())
```

Comment:
this changes previous behavior: in master if we buffer a ledger after ApplyBufferedLedgersWork succeeds, we schedule ApplyBufferedLedgersWork again to apply that ledger. With this change, catchup will finish and will have to wait for another ledger to be re-triggered even if there are no gaps. So I think here if syncing ledgers were not trimmed, we can spin up ApplyBufferedLedgersWork again